src/TableExtension.java

```diff
@@ -92,6 +107,31 @@ public Table(long id) {
       next = StrictMath.max(next, id + 1);
     }
 
+    public Table(java.util.Map<Object,Object> map) {
+      tables.put(this, next);
+      id = next;
+      next++;
+
+      for (java.util.Map.Entry<Object,Object> entry : map.entrySet()) {
+        this.put(entry.getKey(), getTableValue(entry.getValue()));
+      }
+    }
+
+    private Object getTableValue(Object getValue) {
```

Comment:
The `get` part of `getValue` makes me think it's supposed to be a function, we usually try not to use verbs for things that are just data.  Maybe just `value` would work?